examples/aggregator/client/android/java/com/google/oak/aggregator/MainActivity.java

```diff
@@ -106,11 +107,23 @@ public void onClick() {
     EditText thirdValueInput = findViewById(R.id.thirdValueInput);
     values.add(Float.parseFloat(thirdValueInput.getText().toString()));
 
+    // TODO(#988): Move blocking call ouf of UI thread.
     submitSample(bucket, indices, values);
   }
 
-  // TODO(#717): Use Java version of gRPC without invoking JNI.
-  private native void createChannel(String address);
+  /** Gets the custom CA certificate from the assets folder. */
+  private String getCaCertificate() {
+    try {
+      return new BufferedReader(new InputStreamReader(getAssets().open("ca.pem")))
+          .lines()
+          .collect(Collectors.joining("\n"));
+    } catch (Exception exception) {
+      Log.w("Oak", "getCaCertificate", exception);
+      return "";
```

Comment:
I think it would be nicer to change the return type of this method to `Optional<String>`, and the return value here to `Optional.empty()`.

Reply:
Done.